spring-integration-file/src/main/java/org/springframework/integration/file/splitter/FileSplitter.java

```diff
@@ -208,6 +225,11 @@ public void close() throws IOException {
 
 		};
 
+		String firstLineAsHeader =
+				this.firstLieHeaderName != null ?
+						bufferedReader.lines().findFirst().orElse(null)
```

Comment:
Need to be careful here; I think you'll find that the bufferedReader.lines() stream returns an empty string if the file contains just one newline.

Probably easier to avoid the stream here.